autogluon/scheduler/fifo.py

```diff
@@ -19,12 +19,37 @@
 from ..searcher import BaseSearcher
 from ..searcher.searcher_factory import searcher_factory
 from ..utils import save, load, mkdir, try_import_mxboard
+from ..utils.default_arguments import check_and_merge_defaults, \
+    Float, Integer, String, Boolean
 
 __all__ = ['FIFOScheduler']
 
 logger = logging.getLogger(__name__)
 
 
+_DEFAULT_OPTIONS = {
```

Comment:
If the primary purpose of using kwargs is to have these defaults set and the type checks (_CONSTRAINTS), I think we could instead use a decorator function to __init__, with the decorator calling the check_and_merge_defaults function. This output is then passed to the __init__ as input, avoiding the need to alter the input type to be kwargs if desired. See TabularPrediction fit() as an example of this, where we validate and clean the inputs without having to change fit() to be kwargs input.

Reply:
Do you insist on this? Since I am not familiar with decorators, I'd love to not do this now.

Reply:
It is fine for now, I am just mentioning this as an option if things start to get confusing down the road.